src/Symfony/Component/DependencyInjection/Extension/AbstractExtension.php

```diff
@@ -62,4 +67,9 @@ final public function load(array $configs, ContainerBuilder $container): void
 
         $this->executeConfiguratorCallback($container, $callback, $this);
     }
+
+    public function getCompilerPasses(): array
```

Comment:
Would it make sense to use the same signature as we use on the `BundleInterface`?

https://github.com/symfony/symfony/blob/4813d669ff27cc0d1c58a69af9c781b68d25c052/src/Symfony/Component/HttpKernel/Bundle/BundleInterface.php#L39-L46

That allows us to call `$container->addCompilerPass()` from the extension which I believe is a bit easier to understand than the array structure that we're supposed to return here.

Reply:
what about merging compiler passes in ContainerBuilder::merge? then one can register passes from `load()` directly

Reply:
I second @derrabus's idea. A new interface featuring the `public function build(ContainerBuilder $container)` method might be sufficient for adding your custom compiler passes via the extension.

Reply:
what's wrong with ContainerBuilder::merge? build+load seems confusing, or could be avoided maybe?